arch/arm/core/aarch64/switch.S

```diff
@@ -42,7 +42,7 @@ SECTION_FUNC(TEXT, z_arm64_context_switch)
 	stp	x23, x24, [x2], #16
 	stp	x25, x26, [x2], #16
 	stp	x27, x28, [x2], #16
-	stp	x29, x30, [x2], #16
+	stp	x29, xzr, [x2], #16
```

Comment:
`x30` is already saved as part of the `z_arm64_enter_exc` macro and restored when going through `z_arm64_exit_exc`. 

When `ret` in `z_arm64_context_switch` is taken we are back into `z_arm64_svc` for cooperative swapping or `_isr_wrapper` when preemptively swapping. In both cases before leaving these functions we are restoring `x30` as part of `z_arm64_exit_exc`.

`x30` was saved here as part of the process context because for new threads after `ret` we needed to jump into `z_thread_entry_wrapper` but this is not needed anymore.

Reply:
> `x30` is already saved as part of the `z_arm64_enter_exc` macro and restored when going through `z_arm64_exit_exc`.

I see.